### DISCUSSION TITLE
Set up a hardware breakpoint without suspending the underlying application

### DISCUSSION THREAD

**Discussion by kkistm** [eclipse-cdt-cloud/cdt-cloud#36] [Ideas]
This more a discussion point than an idea, but anyway.

As far as I know, GDB inserts breakpoints when the application under debug is resumed. For embedded development it means that to insert a hardware breakpoint the application is suspended first. Some customers might consider as an inconvenience.

An alternative could be to have two connections to a GDB server, and insert breakpoints using an auxiliary connection with `set breakpoint always-inserted on` in action. It might be implemented on IDE level or on CDT Debug Adapter level.

What do you think about it? Is it worth to be implemented (in other words, is there enough amount of customers who are interested in the functionality)? If so, what is the best place to implement it? Any feedback is welcome.

@planger, @jonahgraham

**Comment by jonahgraham** [comment]
> What do you think about it? Is it worth to be implemented (in other words, is there enough amount of customers who are interested in the functionality)? 

I see @MatthewKhouzam has thumbs-upped this, which implies some support for this. I don't know how widely used this will be.

cc: @Kummallinen at Renesas may be interested in this

> If so, what is the best place to implement it? 

I'm not really sure where to implement it. 

> Any feedback is welcome.

The other way to handle this is using non-stop mode which should allow you to insert/remove breakpoints while running.
